enterprise/plugins/com.liferay.ide.server.websphere.core/src/com/liferay/ide/server/websphere/core/ServerStateStartThread.java

```diff
@@ -0,0 +1,197 @@
+ * distribution rights of the Software.
+ */
+
+package com.liferay.ide.server.websphere.core;
+
+import com.liferay.ide.core.util.CoreUtil;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.debug.core.model.IProcess;
+import org.eclipse.wst.server.core.IServer;
+
+/**
+ * @author Simon Jiang
+ */
+
+public class ServerStateStartThread
+{
+
+    // delay before pinging starts
+    private static final int PING_DELAY = 2000;
+
+    // delay between pings
+    private static final int PING_INTERVAL = 250;
+
+    private boolean stop = false;
+    private IProcess mointorProcess;
+    private WebsphereServerBehavior behaviour;
+    private IServer server;
+    private long defaultTimeout = 15 * 60 * 1000;
+    private long timeout = 0;
+    private long startedTime;
+    private IWebsphereServer websphereServer;
+    private ProcessBuilder checkServerStatusProcessBuilder;
+    private String expectedStatusResponse;
+    private URL liferayHomeUrl;
+
+    /**
+     * Create a new PingThread.
+     *
+     * @param server
+     * @param url
+     * @param maxPings
+     *            the maximum number of times to try pinging, or -1 to continue forever
+     * @param behaviour
+     */
+    public ServerStateStartThread( IServer server, WebsphereServerBehavior behaviour )
+    {
+        super();
+        this.server = server;
+        this.behaviour = behaviour;
+        this.mointorProcess = behaviour.getProcess();
+        this.websphereServer = (IWebsphereServer) this.server.loadAdapter( IWebsphereServer.class, null );
+
+        int serverStartTimeout = server.getStartTimeout();
+
+        if( serverStartTimeout < defaultTimeout / 1000 )
+        {
+            this.timeout = defaultTimeout;
+        }
+        else
+        {
+            this.timeout = serverStartTimeout * 1000;
+        }
+
+        liferayHomeUrl = websphereServer.getPortalHomeUrl();
+        IPath profilePath = new Path( websphereServer.getWebsphereProfileLocation() );
+        String websphereServerName = websphereServer.getWebsphereServerName();
+        IPath checkServerStatusCommandPath = null;
+
+        if( CoreUtil.isWindows() )
+        {
+            checkServerStatusCommandPath = profilePath.append( "bin" ).append( "serverStatus.bat" );
+        }
+        else
+        {
+            checkServerStatusCommandPath = profilePath.append( "bin" ).append( "serverStatus.sh" );
+        }
+        List<String> checkServerStatusCommandList =
+            Arrays.asList( checkServerStatusCommandPath.toOSString(), websphereServerName );
+        checkServerStatusProcessBuilder = new ProcessBuilder( checkServerStatusCommandList );
+        expectedStatusResponse = "The Application Server " + "\"" + websphereServerName + "\"" + " is STARTED";
```

Comment:
This i added used to make sure just the liferay portal war failed to start but websphere has been started successfully. 

What's mean just use the ServerStateStopThread?